controllers/serviceclaim_controller.go

```diff
@@ -488,7 +488,34 @@ func (r *ServiceClaimReconciler) DeleteServiceBindingsAndSecret(
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *ServiceClaimReconciler) SetupWithManager(mgr ctrl.Manager) error {
+
+	genPred := predicate.GenerationChangedPredicate{}
+	reconcileOnRegisteredServiceUpdate := func(ctx context.Context, a client.Object) []reconcile.Request {
+		l := log.FromContext(ctx)
+		serviceclaims := &v1alpha1.ServiceClaimList{}
+		opts := &client.ListOptions{}
+		if err := r.List(ctx, serviceclaims, opts); err != nil {
+			return []reconcile.Request{}
+		}
+		rs, ok := a.(*v1alpha1.RegisteredService)
+		if !ok {
+			l.Info("error parsing object to RegisteredService when mapping to ServiceClaim reconciliation trigger", "object", a)
+			return []reconcile.Request{}
+		}
+		reply := make([]reconcile.Request, 0, len(serviceclaims.Items))
+		if rs.Status.State == `Claimed` {
+			for _, sc := range serviceclaims.Items {
+				reply = append(reply, reconcile.Request{NamespacedName: types.NamespacedName{
+					Namespace: sc.Namespace,
+					Name:      sc.Name,
+				}})
+			}
```

Comment:
that's not clear to me. I was expecting something like this

```suggestion
			for _, sc := range serviceclaims.Items {
				if sc.Status != nil && sc.Status.State == "Resolved" && sc.Status.RegisteredService == rs.Name {
					return []reconcile.Request{
						{
							NamespacedName: types.NamespacedName{
								Namespace: sc.Namespace,
								Name:      sc.Name,
							},
						},
					}
				}
			}
			return []reconcile.Request{}
```